src/ResultSet.php

```diff
@@ -3,61 +3,84 @@
 namespace ipl\Orm;
 
 use ArrayIterator;
+use BadMethodCallException;
+use Generator;
 use Iterator;
 use Traversable;
 
+/**
+ * Dataset containing database rows
+ *
+ * @implements Iterator<int, mixed>
```

Comment:
Please avoid such definitions if they don't provide any real benefit. What `TKey` is, can be inferred from `key()` and `TValue` is not mixed, as disussed. If the real type cannot be defined now, we shouldn't try to be explicit while being not explicit at all.